email_sending_domain_certreadrole.tf

```
@@ -10,5 +12,5 @@ module "email_sending_domain_certreadrole" {
   cert_bucket_name = var.cert_bucket_name
   # Certbot stores wildcard certs in a directory with the name of the
   # domain, instead of pre-pending an asterisk.
-  hostname = var.email_sending_domain
+  hostname = each.key
```

We don't have it codified, but I think that `each.value` better represents set values (and it's the same value per [the documentation](https://www.terraform.io/docs/language/meta-arguments/for_each.html#the-each-object)).

```suggestion
  hostname = each.value
```

---

I don't care one way or the other on this, so I'm fine with your suggestion here.